src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -104,7 +104,14 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     }
 
     public canAutoConnect(): boolean {
-        return !hasProfileProperty(this.profile, SHARED_CREDENTIAL_PROPERTIES.MFA_SERIAL) && !this.isSsoProfile()
+        // check if SSO token is still valid
+        if (this.isSsoProfile()) {
+            return new DiskCache().loadAccessToken(this.profile[SHARED_CREDENTIAL_PROPERTIES.SSO_START_URL]!) ===
```

Comment:
can use `!!` as bryce mentioned. if the accesstoken is empty string it's not valid, so don't need a strict `undefined` comparison.